build.py

```diff
@@ -290,15 +290,8 @@ def cmake(self, args):
         self.cmd(f'cmake {" ".join(env_args)} {" ".join(args)}', check_exitcode=True)
 
     def makeinstall(self, target="install"):
-        if target_platform() == "windows":
-            verbose_flag = "" if self._verbose else "-clp:ErrorsOnly"
-            self.cmd(
-                f"msbuild.exe -m:{FLAGS.build_parallel} {verbose_flag} -p:Configuration={FLAGS.build_type} {target}.vcxproj",
-                check_exitcode=True,
-            )
-        else:
-            verbose_flag = "VERBOSE=1" if self._verbose else "VERBOSE=0"
-            self.cmd(f"make -j{FLAGS.build_parallel} {verbose_flag} {target}")
+        verbose_flag = "-v" if self._verbose else ""
+        self.cmd(f"cmake --build . -j{FLAGS.build_parallel} {verbose_flag} -t {target}")
```

Comment:
This omits the `build_type` flag, no? With this change we/users won't be able to set `--build-type=Debug`.

Reply:
I believe we switch code generator in favor of `cmake` if it works then we may have to drop the msbuild installation as well inside the container.

Reply:
Added `--config` to pass the build type.

Reply:
> I believe we switch compiler in favor of cmake if it works then we may have to drop the msbuild installation as well inside the container.

This is mostly a wrapper around `make`, `msbuild`, `gcc`, etc. We still need the underlying dependencies.